### PR TITLE
Upgrade to log4j2.

### PR DESCRIPTION
**Pull request by cstorey**
I notice that in issue #5, you're looking for contributors to adapt the library to log4j2.

One change I'm not sure about is handling some third-party dependencies--I've included them as .jars in this branch, replacing the original source tree.

If there's anything I can do to make it easier to integrate, let me know.

Thanks,

### CODE REVIEW

**Comment by dblock** [comment]
This is good, thanks! It needs a new line (add next release) in https://github.com/dblock/log4jna/blob/master/CHANGELOG.md, please. 

I haven't been actively using this project for a long time, maybe you (@cstorey) or  @tonyniemira are interested in taking it over?

**Comment by dblock** [comment]
The jars are fine. What we need is to mavenize this whole thing I think where dependencies can be declared, and release to maven central too.

**Comment by cstorey** [comment]
I wasn't quite sure when you'd be making the release, so I've left it pending. Unless you'd prefer something else?

I'm afraid that I don't currently have the bandwidth to take this on, really. :/

**Comment by dblock** [comment]
I can merge it, but the build needs to work, you haven't updated the command-line ANT build, please do so, `build all` should succeed from command line.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+x.y (pending)
```
Comment (dblock) [inline CHANGELOG.md]: Lets make this 2.0 (Next), and update the version in version.properties.

**Comment by dblock** [comment]
Please also squash your commits. Thanks!

**Comment by cstorey** [comment]
I usually develop on Linux, so I've got a Rube-Goldberg-eseque network of git repositories to get the code from one place to the other. Unfortunately in this case, I ended up forgetting to pull a bunch of changes.

I've re-verified that those all build and pass the tests on Windows, and squash them as you request. Anything else?

Thanks,

**Comment by dblock** [comment]
Looks and works great, merged.

**Pull request merged**